src/browser/BrowserCodeReader.ts

```diff
@@ -237,16 +240,26 @@ export default class BrowserCodeReader {
         }
     }
 
-    private decodeOnceWithDelay(resolve: (result: Result) => any, reject: (error: any) => any): void {
-        this.timeoutHandler = window.setTimeout(this.decodeOnce.bind(this, resolve, reject), this.timeBetweenScansMillis);
+    private decodeOnceWithDelay(resolve: (result: Result) => any, reject: (error: any) => any, cropFn?: CropFn): void {
+        this.timeoutHandler = window.setTimeout(this.decodeOnce.bind(this, resolve, reject, undefined, undefined, cropFn), this.timeBetweenScansMillis);
     }
 
-    private decodeOnce(resolve: (result: Result) => any, reject: (error: any) => any, retryIfNotFound: boolean = true, retryIfChecksumOrFormatError: boolean = true): void {
+    private decodeOnce(resolve: (result: Result) => any, reject: (error: any) => any, retryIfNotFound: boolean = true, retryIfChecksumOrFormatError: boolean = true, cropFn?: CropFn): void {
         if (undefined === this.canvasElementContext) {
             this.prepareCaptureCanvas();
         }
 
-        this.canvasElementContext.drawImage(this.videoElement || this.imageElement, 0, 0);
+        if (cropFn && this.videoElement) {
+            const videoWidth = this.videoElement.videoWidth;
+            const videoHeight = this.videoElement.videoHeight;
+
+            const { xOffset, yOffset, scanWidth, scanHeight } = cropFn(videoWidth, videoHeight);
+
+            this.canvasElementContext.drawImage(this.videoElement || this.imageElement, xOffset, yOffset, scanWidth, scanHeight, 0, 0, videoWidth, videoHeight);
+
```

Comment:
Can we just sepparate this cropping block into another method (`crop(video, cropFn)`) before mergin'?